mathics/core/definitions.py

```diff
@@ -257,19 +256,30 @@ def clear_definitions_cache(self, name) -> None:
         for k in self.proxy.pop(tail, []):
             definitions_cache.pop(k, None)
 
-    def has_changed(self, maximum, symbols):
-        # timestamp for the most recently changed part of a given expression.
+    def uncertain_final_value(self, last_evaluated_time: int, symbols: set) -> bool:
```

Comment:
That returns a bool, so maybe `is_uncertain_final_value` be clearer.

Reply:
@TiagoCavalcante  thanks - I will change this in the next commit.